src/libgit2/config.c

```diff
@@ -606,8 +607,19 @@ static int get_backend_for_use(git_config_backend **out,
 
 	git_vector_foreach(&cfg->backends, i, backend) {
 		if (!backend->backend->readonly) {
-			*out = backend->backend;
-			return 0;
+			switch (use) {
+			case BACKEND_USE_SET:
+				*out = backend->backend;
+				return 0;
+			case BACKEND_USE_DELETE:
+				/* return the first backend that contains the given config-name */
+				backend->backend->get(backend->backend, name, &entry);
```

Comment:
Do I need to lock/unlock the backend here? Is there a risk of data getting written out from underneath me?